libhoney.go

```diff
@@ -636,42 +723,25 @@ func (e *Event) SendPresampled() (err error) {
 		return errors.New("No Dataset for Honeycomb. Can't send datasetless.")
 	}
 
-	txOnce.Do(func() {
-		if tx == nil {
-			tx = &txDefaultClient{
-				maxBatchSize:         DefaultMaxBatchSize,
-				batchTimeout:         DefaultBatchTimeout,
-				maxConcurrentBatches: DefaultMaxConcurrentBatches,
-				pendingWorkCapacity:  DefaultPendingWorkCapacity,
-			}
-			tx.Start()
-		}
-	})
-
-	tx.Add(e)
-	return nil
-}
-
-// sendResponse sends a dropped event response down the response channel
-func sendDroppedResponse(e *Event, message string) {
-	r := Response{
-		Err:      errors.New(message),
-		Metadata: e.Metadata,
+	e.client.ensureTransmission()
+	txEvent := &transmission.Event{
```

Comment:
seems unfortunate to have to duplicate the event struct and copy everything over. Assuming this was to prevent circular imports? Could we eventually refactor event into its own package?

Reply:
We could, but not easily. The public interface of the SDK exports `libhoney.Event` as a thing, so I need to maintain that. We could move Event to its own package and then have libhoney.Event include that type (in the same way there's a `transitionalOutput` type), but it gets messy with the fieldholder. This diff is big enough; I'd rather keep it as is for now and try and move it later if it remains an issue.